tests/jballerina-unit-test/src/test/resources/test-src/statements/dostatement/do-stmt.bal

```diff
@@ -95,6 +101,28 @@ function testReturnWithinOnFail() returns string  {
    return str;
 }
 
+function testOnFailWithoutVariable() returns string  {
+   string str = "";
+   do {
+     error err = error("custom error", message = "error value");
+     str += "Before failure throw";
```

Comment:
Let's not use terms like throw. We don't throw errors in b7a, we panic. In these tests, we aren't even panicking.